modules/snooker/src/main/ts/ephox/snooker/api/Generators.ts

```diff
@@ -166,12 +171,55 @@ const merging = function (generators: Generators) {
       // Remove any width calculations because they are no longer relevant.
       Css.remove(raw, 'width');
       Css.remove(cell, 'width');
+
+      Attribute.getOpt(cell, 'scope').each((scope) => Attribute.set(raw, 'scope', scope));
```

Comment:
You should be able to re-use the scope attribute in the outers scope here I believe. That'd save the duplicate code.

- [ ] Look at removing duplicate scope look up code here.

Reply:
There's still some duplication here, as you have the 2 following bits of code that are the same:
```
const scope = Attribute.getOpt(cell, 'scope').map(
  (attribute) => attribute.substr(0, 3) // Changes colgroup to col and rowgroup to row.
);
```
and
```
const attributeOpt = Attribute.getOpt(cell, 'scope');

return attributeOpt.map((attribute) => attribute.substr(0, 3)); // Attribute can be col, colgroup, row, and rowgroup.
```